proposed/2022/Additional-Metadata-Attributes.md

```diff
@@ -0,0 +1,76 @@
+# Additional Metadata Attributes
+
+- Author Name [@redth](https://github.com/redth)
+- Start Date (2022-10-190)
+- GitHub Issue (GitHub Issue link)
+- GitHub PR (GitHub PR link)
+
+## Summary
+
+Allow the inclusion of additional metadata properties in package authoring and allow them to be used in search queries.
+
+## Motivation 
+
+MAUI (and previously Xamarin) has an ecosystem of 'binding' library NuGet packages which surface native platform library usage within apps through C# API projections.  These native platform libraries are often available on their native package management system counterparts (ie: Maven, CocoaPods, etc.).  Typically binding nuget packages redistribute the native libraries within them.
+
+It's currently challenging/impossible to discern which NuGet packages map to a given native platform library package identity.  
+
+Moving the ecosystem forward, it would be very helpful to be able to programmatically determine the native platform library identities to:
+
+1. Deduplicate inclusion of packages through transitive dependencies (which often result in native toolchain build errors if not resolved)
+2. Discover eligible packages to fulfill dependencies of known native platform libraries (ie: integrating the output of a gradle build and automatically matching up / suggesting existing packages for dependencies)
+
+Identity can consist of multiple attributes, for example a Maven package has:
+
+- Group Id (eg: `com.company.product`)
+- Artifact Id (eg: `ProductSdk`)
+- Version (eg: `1.3.0`)
+
+While a useful consumer of the search service might query by (GroupId='com.company.product' && ArtifactId='ProductSdk') in order to return a list of packages to further inspect version attributes of to determine if any package identities satisfy its requirements.
+
+## Explanation
+
+The NuGet search API already allows the specification of various package [metadata fields to search by in the query parameter](https://learn.microsoft.com/en-us/nuget/consume-packages/finding-and-choosing-packages#search-syntax).  This proposal is simply an extension of that existing query syntax to include additional, potentially arbitrary attributes both in the .nuspec format as well as the search query.
+
+### Functional explanation
```

Comment:
Do we have any considerations beyong discoverability? 
Maybe something specific to management some of these related packages within your project or is that not a big concern?

Reply:
> Do we have any considerations beyong discoverability? 

1:1 mapping of native (maven or native lib) to nuget would make

- security checks easier
- curation (optional) easier

> Maybe something specific to management some of these related packages within your project or is that not a big concern?

We do that, but formal/standardized/central method would help, both us and (IMO) nuget team.

Reply:
Now thinking a bit deeper 1:1 is oversimplification. In cross platform scenario there will be artifact per platform (Android, iOS) and sometimes multiple artifacts per platform.